package.json

```diff
@@ -92,7 +92,13 @@
       "debug/variables/context": [
         {
           "command": "memory-inspector.show-variable",
-          "when": "memory-inspector.canRead"
+          "when": "canViewMemory && memory-inspector.canRead"
+        }
+      ],
+      "view/item/context": [
+        {
+          "command": "memory-inspector.show-variable",
+          "when": "canViewMemory && memory-inspector.canRead"
```

Comment:
Since this view plugin doesn't contribute a view to which this could apply, this may belong in the plugins that implement tree views with viewable items?

Reply:
This one is a tough call. Other extensions can create generic views (e.g. we have registers split out in our debugger), however the VS Code system can determine that the hexedit view can be opened using the `canViewMemory` context.

At some point a soft dependency needs to exist between extensions (one extension needs to know of another to extend the context menus).

Our debugger doesn't need to extend the memory inspector functionality (it uses c expressions) and others may have a similar situation, so it felt cleaner and more concise to key this for any views that VS Code determines can open memory in a central location.

Reply:
I think that's fair. One context key is from VSCode, and the other is from this plugin, and the menu contribution is VSCode, so it's not like this registration is making a lot of assumptions beyond those supported by VSCode itself. Might as well give people options they didn't know they had :-).